builder/builder.go

```diff
@@ -836,6 +839,58 @@ func (b *Builder) ListBundles(listType listType, tree bool) error {
 	return nil
 }
 
+var validEditors = map[string]string{
+	"joe":   "/usr/bin/joe",
+	"emacs": "/usr/bin/emacs",
+	"ctags": "/usr/bin/ctags",
+	"nano":  "/usr/bin/nano",
+	"vim":   "/usr/bin/vim",
+}
+
+//EditBundles prints out a bundle list in either a flat list or tree view
+func (b *Builder) EditBundles(bundles []string, noEdit bool, editor string, add bool) error {
+	if editor != "" {
+		b.BundleEditor = editor
+	}
+	editorCmd, exists := validEditors[b.BundleEditor]
+	if !exists {
+		return errors.Errorf("Invalid editor choice: %s", b.BundleEditor)
+	}
+
+	for _, bundle := range bundles {
+		path, err := b.getBundlePath(bundle)
+		if err != nil {
+			return err
+		}
+
+		if !b.isLocalBundle(path) {
+			localPath := filepath.Join(b.LocalBundleDir, bundle)
+			if err := helpers.CopyFile(localPath, path); err != nil {
+				return err
+			}
+			path = localPath
+		}
+
+		if !noEdit {
+			if err := helpers.RunCommand(editorCmd, path); err != nil {
+				return errors.Wrapf(err, "Failed to edit bundle: %s", bundle)
```

Comment:
This is high risk. Some editors return a non-zero exit code if some commands fail, which might be as simple as a search failing.

Reply:
That's a great point.  In theory, I don't _really_ care what the editor returned, I just care if I can read/parse it afterward. I should probably just ignore the error here, and let the parsing trigger the error.